File: cplusplus/design_pattern/builderProj/Builder2.cpp
#include "stdafx.h"
#include "Builder2.h"


Builder2::Builder2()
{
    //std::cout << "Builder2::Builder2()" << std::endl;
}


Builder2::~Builder2()
{
   // std::cout << "Builder2::~Builder2()" << std::endl;
}


Builder2::Builder2(std::shared_ptr<IBuilderProduct> product) : IBuilder(product)
{
    //std::cout << "Builder2::Builder2(std::shared_ptr<IBuilderProduct> product)" << std::endl;
}


void Builder2::createPart1()
{
    getProduct()->product_part1();
}


void Builder2::createPart2()
{
    getProduct()->product_part2();
}

File: Java/pattern/builder/src/com/pattern/Builder2.java
package com.pattern;

public class Builder2 extends IBuilder {
	public Builder2(IProduct myProduct) {
		super();
		this.myProduct = myProduct;
	}

	private IProduct myProduct;

	@Override
	public void createPart1() {
		myProduct.product_part1();

	}

	@Override
	public void createPart2() {
		myProduct.product_part2();

	}

}
